Riot/Modules/Settings/SignOut/SignOutFlowPresenter.swift

```diff
@@ -0,0 +1,162 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+@objc protocol SignOutFlowPresenterDelegate {
```

Comment:
I don't mind this approach with independent protocol methods, though I thought we mostly use ViewModels/ViewStates with individual .loading, .loaded, .error cases, and single method to dispatch them. I am not suggesting to change it, just wanted to see what you thought about those two approaches

Reply:
So I started with a callback and the cases you described as I'd much prefer that. But then I needed to switch to the delegate when integrating the presenter into the `SettingsViewController` as the `.error` case couldn't have an associated value and also be compatible with objc.